sklearn/decomposition/_dict_learning.py

```diff
@@ -490,6 +490,26 @@ def _update_dict(
         print(f"{n_unused} unused atoms resampled.")
 
 
+@validate_params(
+    {
+        "X": ["array-like"],
+        "n_components": [Interval(Integral, 1, None, closed="left")],
+        "alpha": [Interval(Real, 0, None, closed="left")],
```

Comment:
According to the docstring, `alpha` should be an int. However, it works with float and looking at the code, it is passed to `sparse_encode`, whose [docs](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.sparse_encode.html) say it can be float, so presumably the docstring is incorrect here. So it should probably be amended.